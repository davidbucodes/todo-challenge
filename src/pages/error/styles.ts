import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div``;
  export const Title = styled.h2``;
  export const Details = styled.div``;
}
