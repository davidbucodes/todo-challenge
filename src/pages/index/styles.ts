import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div`
    display: flex;
    flex-direction: column;
    gap: 1ch;
  `;
  export const Loading = styled.div``;
  export const LoadingError = styled.div``;
}
