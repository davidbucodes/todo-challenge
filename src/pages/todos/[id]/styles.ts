import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div``;
  export const TodoDetail = styled.div`
    padding-bottom: 1ch;
  `;
  export const TodoDetailLoading = styled.div``;
  export const TodoDetailError = styled.div``;
  export const TodoDetailKeyValue = styled.div`
    display: flex;
    gap: 1ch;
  `;
  export const Key = styled.span`
    font-weight: bold;
  `;
  export const Value = styled.span`
    flex-grow: 1;
  `;
}
