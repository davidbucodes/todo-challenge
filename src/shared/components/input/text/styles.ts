import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div``;
  export const TextInput = styled.input.attrs({
    type: "text",
  })``;
  export const InputLabel = styled.label``;
}
