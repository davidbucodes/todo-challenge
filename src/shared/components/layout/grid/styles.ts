import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div``;
  export const Grid = styled.ul`
    display: flex;
    gap: 20px;
    flex-wrap: wrap;
    justify-content: center;
    margin-block-start: 0;
    margin-block-end: 0;
    padding-inline-start: 0;
  `;
  export const GridItem = styled.li`
    display: inline-flex;
  `;
}
