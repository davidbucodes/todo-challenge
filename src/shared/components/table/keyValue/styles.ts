import styled from "styled-components";
export namespace Styles {
  export const Wrapper = styled.div``;
  export const Table = styled.div``;
  export const KeyValuePair = styled.div`
    display: flex;
    gap: 1ch;
    padding-bottom: 1ch;
  `;
  export const Key = styled.span`
    font-weight: bold;
  `;
  export const Value = styled.span`
    flex-grow: 1;
  `;
}
