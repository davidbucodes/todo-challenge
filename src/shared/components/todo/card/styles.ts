import styled from "styled-components";

export namespace Styles {
  export const Wrapper = styled.div<{ isCompleted: boolean }>`
    width: 250px;
    text-align: center;
    border: 1px solid black;
    background-color: ${({ theme, isCompleted }) =>
      isCompleted
        ? theme.colors.background.green
        : theme.colors.background.crimson};
  `;
  export const Id = styled.div`
    padding: 20px 0;
  `;
  export const Title = styled.div`
    padding-bottom: 40px;
  `;
}
