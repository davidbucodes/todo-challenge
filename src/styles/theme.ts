export const theme = {
  colors: {
    background: {
      green: "green",
      crimson: "crimson",
    },
  },
};

type CustomTheme = typeof theme;

declare module "styled-components" {
  export interface DefaultTheme extends CustomTheme {}
}
