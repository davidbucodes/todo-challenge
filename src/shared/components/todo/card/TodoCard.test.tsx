import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import { theme } from "../../../../styles/theme";
import * as stories from "./TodoCard.stories";

const { Default, Completed, LongTitle, NotCompleted, ShortTitle } =
  composeStories(stories);

describe("TodoCard", () => {
  describe("Default", () => {
    it("should render correctly", () => {
      const { container } = render(<Default />);
      const idElement = screen.getByText(`#${Default.args.todo!.id!}`);
      const titleElement = screen.getByText(Default.args.todo!.title!);
      expect(container).toMatchSnapshot();
      expect(idElement).toBeInTheDocument();
      expect(titleElement).toBeInTheDocument();
    });
  });

  describe("ShortTitle", () => {
    it("should render correctly", () => {
      const { container } = render(<ShortTitle />);
      const idElement = screen.getByText(`#${ShortTitle.args.todo!.id!}`);
      const titleElement = screen.getByText(ShortTitle.args.todo!.title!);
      expect(container).toMatchSnapshot();
      expect(idElement).toBeInTheDocument();
      expect(titleElement).toBeInTheDocument();
    });
  });

  describe("LongTitle", () => {
    it("should render correctly", () => {
      const { container } = render(<LongTitle />);
      const idElement = screen.getByText(`#${LongTitle.args.todo!.id!}`);
      const formattedTitle = `${LongTitle.args.todo!.title!.substring(
        0,
        20
      )}...`;
      const titleElement = screen.getByText(formattedTitle);
      expect(container).toMatchSnapshot();
      expect(idElement).toBeInTheDocument();
      expect(titleElement).toBeInTheDocument();
    });
  });

  describe("NotCompleted", () => {
    it("should render correctly", () => {
      const { container } = render(<NotCompleted />);
      const card = container.firstChild?.firstChild;
      expect(container).toMatchSnapshot();
      expect(card).toHaveStyle({
        "background-color": theme.colors.background.crimson,
      });
    });
  });

  describe("Completed", () => {
    it("should render correctly", () => {
      const { container } = render(<Completed />);
      const card = container.firstChild?.firstChild;
      expect(container).toMatchSnapshot();
      expect(card).toHaveStyle({
        "background-color": theme.colors.background.green,
      });
    });
  });
});
