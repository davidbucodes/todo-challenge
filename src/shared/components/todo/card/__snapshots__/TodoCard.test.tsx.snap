// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard Completed should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi izpPZC"
    >
      <div
        class="sc-ewnqHT bDrKue"
      >
        #
        1
      </div>
      <div
        class="sc-fFGjHI bmMTuD"
      >
        delectus aut autem
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoCard Default should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi hjhHBO"
    >
      <div
        class="sc-ewnqHT bDrKue"
      >
        #
        1
      </div>
      <div
        class="sc-fFGjHI bmMTuD"
      >
        delectus aut autem
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoCard LongTitle should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi hjhHBO"
    >
      <div
        class="sc-ewnqHT bDrKue"
      >
        #
        1
      </div>
      <div
        class="sc-fFGjHI bmMTuD"
      >
        long title long titl...
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoCard NotCompleted should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi hjhHBO"
    >
      <div
        class="sc-ewnqHT bDrKue"
      >
        #
        1
      </div>
      <div
        class="sc-fFGjHI bmMTuD"
      >
        delectus aut autem
      </div>
    </div>
  </div>
</div>
`;

exports[`TodoCard ShortTitle should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <div
      class="sc-gTRrQi hjhHBO"
    >
      <div
        class="sc-ewnqHT bDrKue"
      >
        #
        1
      </div>
      <div
        class="sc-fFGjHI bmMTuD"
      >
        short title
      </div>
    </div>
  </div>
</div>
`;
